src/giftCards/GiftCardSettings/messages.ts

```diff
@@ -0,0 +1,33 @@
+import { GiftCardSettingsErrorFragment } from "@saleor/fragments/types/GiftCardSettingsErrorFragment";
+import { commonMessages } from "@saleor/intl";
+import { GiftCardSettingsErrorCode } from "@saleor/types/globalTypes";
+import commonErrorMessages from "@saleor/utils/errors/common";
+import { defineMessages, IntlShape } from "react-intl";
+
+export const giftCardSettingsPageMessages = defineMessages({
+  title: {
+    defaultMessage: "Gift Cards Settings",
+    description: "header"
+  }
+});
+
+export function getGiftCardSettingsErrorMessage(
+  error: GiftCardSettingsErrorFragment | undefined,
+  intl: IntlShape
+): string {
+  if (error) {
+    switch (error.code) {
+      case GiftCardSettingsErrorCode.GRAPHQL_ERROR:
+        return intl.formatMessage(commonErrorMessages.graphqlError);
+      case GiftCardSettingsErrorCode.REQUIRED:
+        return intl.formatMessage(commonMessages.requiredField);
+      case GiftCardSettingsErrorCode.INVALID:
+        return intl.formatMessage(commonErrorMessages.invalid);
+
+      default:
+        return intl.formatMessage(commonErrorMessages.unknownError);
+    }
+  }
+
+  return undefined;
+}
```

Comment:
I feel like we're redefining this function many times with the exact same messages defined. I'd suggest to do a an error message extractor for all the generic ones but leave the specific functions in order to extend errors when needed:
```suggestion
export function getGiftCardSettingsErrorMessage(
  error: GiftCardSettingsErrorFragment | undefined,
  intl: IntlShape
): string {
  if (error.code) {
    // and here we could extend the messages to giftcard specific when/if needed
    return getGenericFormFieldErrorMessage(error.code);
  }
  
  return undefined;
}
```

Reply:
The types are different, here `GiftCardSettingsErrorCode` but in other place `GiftCardErrorCode`, this is why function with switch case is redefined.

Reply:
I've added `CommonErrorCode` enum to represent common schema enums members.
```typescript
enum CommonErrorCode {
  GRAPHQL_ERROR = "GRAPHQL_ERROR",
  INVALID = "INVALID",
  REQUIRED = "REQUIRED"
}
interface CommonError<ErrorCode> {
  code: ErrorCode | CommonErrorCode;
  field: string | null;
}

export function getCommonFormFieldErrorMessage<ErrorCode>(
  error: CommonError<ErrorCode> | undefined,
  intl: IntlShape
): string {
  if (error) {
    switch (error.code) {
      case CommonErrorCode.GRAPHQL_ERROR:
        return intl.formatMessage(commonErrorMessages.graphqlError);
      case CommonErrorCode.REQUIRED:
        return intl.formatMessage(commonMessages.requiredField);
      case CommonErrorCode.INVALID:
        return intl.formatMessage(commonErrorMessages.invalid);

      default:
        return intl.formatMessage(commonErrorMessages.unknownError);
    }
  }

  return undefined;
}
```

Reply:
I like it but there's no need to have those keys as enums, use string unions:
```
type CommonErrorCode = "GRAPHQL_ERROR" | ....
getCommonFormFieldErrorMessage(error: CommonErrorCode, intl: IntlShape)
```